### PR TITLE
Notification integration update

### LINKED ISSUE #9: Report of anomalies within the app
As a technician, I want to be able to see an issue report so that I can see the details of an issue

**Acceptance Criteria**

 - Report has to be within the application
 - Contains specific details regarding what went wrong. This may include the values being pushed by the machines, the amount of time the defect stayed for, which machine (maybe an ID) had the defect.
 - Report has to be linked by the notification
 - Only one report should be created for the duration of the defect
 - The user should be able to acknowledge the reports to turn notifications for it off.
 - The user should be able to indicate a sensor has been fixed and is now working to turn notifications back on.

**Is your feature request related to a problem? Please describe.**
Users should be able to see a detailed report of the issue that they were notified of by the notification system.

**Describe the solution you'd like**
A detailed report within the app, which describes the issue, giving different information such as the faulty machine, the values that were faulty and the amount of time that the defect stayed for.
This kind of report needs to be both accessible through the notifications (like a link), and through the app itself.

**Describe alternatives you've considered**
N/A
**Additional context**
Child of #5

### PR DESCRIPTION
**Pull request by George-E-M**
## GitHub Issue Solved:

closes #9 

## Current behaviour

There is no indication regarding acknowledging a sensor issue and getting rid of the notifications within the application.

## Changed behaviour

The user can now turn off notifications for a sensor by acknowledging the issues with it. They cant then turn the notifications back on by indicating the sensor is now fixed.

## PR checklist

Remember to check the following

 - [ ] Tag specific people to review your PR
 - [x] Update the ReadMe with any new run instructions
 - [x] Closes the associated issue (if applicable)

## Notes

You will need to create an .env file within the api/functions folder. Within this file you will need the sendgrid api key and a variable specifying the node environment is development. The file will look like the following (in addition to any thing else you have in the file).
```
SENDGRID_API_KEY=<Sendgrid API Key>
NODE_ENV=development
```
To get the sendgrid API key refer to the pinned message in discord backend chat.

### Run instructions

Run npm start from the root of the project. From the machines page navigate to the "TestingPR" machine and from within that machine navigate to the ":(" sensor. Within this sensor there will be an indication of the notification issues. If you acknowledge it the message should be replaced with one prompting you to indicate the machine has been fixed. Once you press that there should be no message.  Also note this sensor is specified because it has the notificationStatus field and that field is set to Unacknowledged.

In addition to this, the notification service was setup to work with the ingestor. This means if you run the ingestor file
```
ts-node src/ingestor/ingestor.ts
```
from the api/functions directory you should see the sensor hardcoded into the project have its notification status change on the firestore to Unacknowledged.

Finally, the function to handle the regular notification to remind users of the machine errors can be tested. by adding the line 
```
updateUsers();
```
to the notificationService.ts file (adding it on line 11 works) and then running that file using
```
ts-node src/notifications/notificationService.ts
```
in order to check this is working just log the emailMsg variable to see the response is being generated correctly (the sendgrid email cap may have been reached so the send() line is currently commented out).

### CODE REVIEW

File: api/functions/src/index.ts [hunk 1]
```diff
@@ -2,12 +2,17 @@
...
 import { ConstructFileDownloadServer } from './downloads/server';
 
 const USCentralRegion = functions.SUPPORTED_REGIONS[0];
 exports.graph = functions
   .region(USCentralRegion)
   .https.onRequest(ConstructGraphQLServer());
+// exports.notify = functions.pubsub
+//   .schedule('0 * * * *')
+//   .timeZone('Pacific/Auckland')
+//   .onRun(updateUsers);
```
Comment (FrankWangma) [inline api/functions/src/index.ts:15]: Do we need this commented out code?

File: api/functions/src/ingestor/thresholdDetection.ts [hunk 2]
```diff
@@ -31,8 +36,22 @@ export function doThresholdDetection(rmsValue: number) {
           state.thresholdValue +
           ' met, notification being sent.'
```
Comment (FrankWangma) [inline api/functions/src/ingestor/thresholdDetection.ts:37]: Not your fault, but should the "notification being sent" message only be printed out after passing 10 bad readings?

**Review by FrankWangma — CHANGES_REQUESTED** [review]
Everything's looking good for the most part. Just added a few comments which need to be looked at.

File: api/functions/src/index.ts (hunk 1 above)
Comment (George-E-M) [reply api/functions/src/index.ts:15]: No, will uncomment it now.

Comment (George-E-M) [reply api/functions/src/index.ts:15]: Wait, we need the code but it doesn't need to be commented out

File: api/functions/src/ingestor/thresholdDetection.ts (hunk 2 above)
Comment (George-E-M) [reply api/functions/src/ingestor/thresholdDetection.ts:37]: Probably but thats just test code not actual production stuff so we can just leave it

**Review by FrankWangma — APPROVED** [review]
LGTM

**Pull request merged**